.github/workflows/dotnet.yaml

```diff
@@ -8,22 +8,56 @@ on:
       - "dotnet/**"
       - ".github/workflows/dotnet.yaml"
       - "install_tooling.sh"
+  workflow_call:
+    secrets:
+      IMAGE_PREFIX:
+        description: "registry.com/loftsh in registry.com/loftsh/language:TAG"
+        required: true
 
 env:
-  IMAGE_PREFIX: loftsh
+  IMAGE_PREFIX: ${{ secrets.IMAGE_PREFIX }}
+  LANGUAGE: "dotnet"
+  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+
+defaults:
+  run:
+    working-directory: dotnet
 
 jobs:
   dotnet:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - tag: 6.0-alpine
+            target_tag: latest
+          - tag: 6.0-alpine
+            target_tag: 6.0
+          - tag: 6.0-alpine
+            target_tag: 6.0-alpine
+          - tag: 5.0-bullseye-slim
+            target_tag: 5.0
+          - tag: 5.0-bullseye-slim
+            target_tag: 5.0-bullseye-slim
+          - tag: 3.1
+            target_tag: 3.1
```

Comment:
I don't think we should try to cover all of these different distros. This will be really hard to maintain for all languages.

Reply:
indeed, I reduced the versions/flavours